src/main/java/uk/ac/cam/cl/dtg/util/ReflectionsUtil.java

```diff
@@ -0,0 +1,53 @@
+package uk.ac.cam.cl.dtg.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ReflectionsUtil {
+  public static List<Class<?>> getClasses(String packageName) throws ClassNotFoundException, IOException,
```

Comment:
One thing you could consider in order to get rid of these checked exceptions is to catch them in this method, and then decide what to do with them, individually. For instance, you could choose to re-throw them as an unchecked exception instead, like this:

```java
try {
  classes.addAll(findClasses(directory, packageName));
} catch (ClassNotFoundException e) {
  throw new RuntimeException(e);
}
```

This will allow you to get rid of this `throws ClassNotFoundException, IOException, URISyntaxException` both here and wherever this function is used.

Another alternative to re-throwing the exception could be to log an error and move on. It's up to you, whatever you think it's best.